rfcs/0031-variable-length-header-field/0031-variable-length-header-field.md

```diff
@@ -7,30 +7,29 @@ Organization: Nervos Foundation
+Many extensions require adding new fields to the block data structure, for example, flyclient requires a 64-byte hash in the header. However, there are not enough reserved bits for this. There are workarounds such as storing these data in the cellbase transaction, but this has a heavy overhead for clients that want to verify the chain using only PoW. Because they have to download the cellbase transaction and the merkle tree proof of the cellbase transaction, which can be larger than the block header itself.
 
-This document proposes a solution to add a variable length field in the block. How to interpret the new field is beyond the scope of this document and must be defined and deployed via a future soft fork. Although the field is added to the block body, nodes can synchronize the block header and this field together in the future version.
+This document proposes a solution to add a variable length field in the block data structure. The interpretation of the new field is beyond the scope of this document and will need to be defined and deployed in a future soft fork. Although the field is added to the block body, nodes can synchronize the block header and this field together in the future version.
 
 ## Specification
 
 The block header is encoded as a molecule struct, which consists of fixed length fields. The header binary is just the concatenation of all the fields in sequence.
 
-There are many ways to add the variable length field to the block header. This RFC proposes to replace the `uncles_hash` in the header with the new field `extra_hash`, which is also a 32-byte hash. The block will have a new field `extension`.
+There are many ways to add the variable length field to the block header. This RFC proposes to replace the `uncles_hash` field in the header with the new `extra_hash` field, which is also a 32-byte hash. The block data structure will have a new `extension` field.
 
 There are two important time points to deploy this RFC, activation epoch A and extension application epoch B.
 
-In blocks before epoch A, the `extension` must be absent. The value of `extra_hash` is the same as the original `uncles_hash` in these blocks, so this RFC will not change the serialized headers of existing blocks. The field `extra_hash` is all zeros when the `uncles` field is empty, or `ckbhash` on all the uncle header hashes concatenated together.
+In blocks before epoch A, the `extension` field must be absent. The value of `extra_hash` is the same as the original `uncles_hash` in these blocks, so this RFC will not change the serialized headers of existing blocks. If the `uncles` field is empty, the `extra_hash` field is zero; otherwise, it is `ckbhash`, which is the concatenation of all the uncle header hashes.
```

Comment:
If the `uncles` field is empty, the `extra_hash` field is zero; otherwise, the ckbhash function is applied to the concatenation of all uncle header hashes.